actions/lavamoat-policy-differ/src/analyze.js

```diff
@@ -0,0 +1,200 @@
+/**
+ * @param {string} id
+ * @returns {string}
+ */
+const id2name = (id) => id.split('>').pop() || id
+
+/** 
+ * @returns {Record<string, { [key: string]: string[] }>} 
+ */
+const create = () => ({})
+
+/**
+ * @param {string[]} keys
+ * @param {string} action
+ * @returns {Record<string, { [key: string]: string[] }>}
+ */
+const aggregateAction = (keys, action) =>
+  keys.reduce((aggr, key) => {
+    const name = id2name(key)
+    if (!aggr[name]) {
+      aggr[name] = { [action]: [] }
+    }
+    aggr[name][action].push(key)
+    return aggr
+  }, create())
+
+/**
+ * @param {string[]} addedKeys
+ * @param {string[]} removedKeys
+ * @param {string[]} currentKeys
+ */
+const aggregateChangesByPackageName = (addedKeys, removedKeys, currentKeys) => {
+  const addedPackages = aggregateAction(addedKeys, 'added')
+  const removedPackages = aggregateAction(removedKeys, 'removed')
+  const currentPackages = aggregateAction(currentKeys, 'current')
+
+  const keysOfInterest = new Set([
+    ...Object.keys(addedPackages),
+    ...Object.keys(removedPackages),
+  ])
+
+  const aggr = create()
+
+  for (const key of keysOfInterest) {
+    aggr[key] = {
+      added: addedPackages[key]?.added || [],
+      removed: removedPackages[key]?.removed || [],
+      current: currentPackages[key]?.current || [],
+      unchanged: (currentPackages[key]?.current || []).filter(
+        (x) => !addedPackages[key]?.added.includes(x)
+      ),
+    }
+  }
+
+  return aggr
+}
+
+/**
+ * 
+ * @param {string} str 
+ */
+const c = (str) => '`' + str + '`'
+
+/**
+ * @param {string[]} keys
+ * @param {import('lavamoat-core').Resources} resources
+ */
+const findWhatDependsOn = (keys, resources) =>
+  Object.entries(resources)
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    .filter(([_, policy]) =>
+      keys.some((key) => Object.keys(policy?.packages || {}).includes(key))
```

Comment:
would recommend pulling stuff like `Object.keys(policy.packages)` into its own variable in scope--and maybe using a `Set`--since that can get biggish, yes?